shared.ts

```diff
@@ -1,12 +1,14 @@
 import fc from "fast-check";
 import {
-  BaseType,
+  BaseTypeAfterEnrich,
```

Comment:
Could be at least EnrichedType? or EnrichedBaseType? In case we go with "enriched".